packages/user/Supervisor/ui/src/main.ts

```diff
@@ -27,12 +27,21 @@ const callHandlers: CallHandler[] = [];
 const shouldHandleMessage = (message: MessageEvent) => {
     const fromTop = message.source == window.top;
     const fromParent = message.source == window.parent;
-    const protocol = new URL(message.origin).protocol + "//";
-    const urlSuffix = siblingUrl().slice(protocol.length);
-    const fromSameOrigin = message.origin.endsWith(urlSuffix);
+
+    const messageUrl = new URL(message.origin);
+    const messageHasSubdomain = messageUrl.hostname.split(".").length > 2;
```

Comment:
Won't this always be true?

Reply:
I was thinking if the message comes from the homepage